src/devices/bus/coco/coco_t4426.cpp

```diff
@@ -0,0 +1,195 @@
+// license:BSD-3-Clause
+// copyright-holders:Joakim Larsson Edstrom
+/***************************************************************************
+ *
+ *  coco_t4426.c
+ *
+ *  Terco T4426 CNC Programming Station multi cart
+ *
+ *  The code here is heavily inspired by coco_pak and coco_232
+ *
+ *  +-------------------------------------------------------------------------------+
+ *  ||__|+-----+    oo   75                               |O ||||||||||||||| O|     |
+ *  |XTL||MC   |    oo  110                                                         |
+ *  |1.8||14411|    oo  ..        +--+ +--+                                         |
+ *  |432||BRG  |    ==  600       |74| |74| +------+                                |
+ *  |MHz||     |    oo  ..        |LS| |LS| |MC1488|                                |
+ *  +---+|     |    oo 7200       |139 |00| +------+                                |
+ *  |    +-----+    oo 9600       +--+ +--+                    +--+                 |
+ *  |                                      +-------------+     |MC|                 |
+ *  |    +-----+   +-----+    +-----+      | EF68B50P    |     |14|                 |
+ *  |    | 2764|   | 2764|    |     |      | ACIA        |     |89|                 |
+ *  |    |     |   |     |    |     |      +-------------+     +--+                 |
+ *  |    |CA   |   |CA   |    |PMOS |                                               |
+ *  |    | 4426|   | 4426|    | 4426|   +-------------------+                       |
+ *  |    |  -6 |   |  -7 |    |     |   |  EP68B21P         |                       |
+ *  |    |     |   |     |    |     |   |  PIA              |                       |
+ *  |    |     |   |     |    +-----+   +-------------------+                       |
+ *  |    +-----+   +-----+                                                          |
+ *  |    +-----+   +-----+    +-----+   +-----+    +-----+   +-----+                |
+ *  |    | 2764|   | 2764|    | 2764|   | 2764|    | 2764|   | 2764|                |
+ *  |    |     |   |     |    |     |   |     |    |     |   |     |                |
+ *  |    |CA   |   |CA   |    |PD   |   |PD   |    |ED   |   |ED   |                |
+ *  |    | 4426|   | 4426|    | 4426|   | 4426|    | 4426|   | 4426|                |
+ *  |    |  -5 |   |  -4 |    |  -3 |   |  -2 |    |  -1 |   |  -0 |                |
+ *  |    |     |   |     |    |     |   |     |    |     |   |     |       OO       |
+ *  |    |     |   |     |    |     |   |     |    |     |   |     |                |
+ *  |    +-----+   +-----+    +-----+   +-----+    +-----+   +-----+                |
+ *  +-------------------------------------------------------------------------------+
+ *
+ ***************************************************************************/
+
+#define VERBOSE 0
+
+#define LOGPRINT(x) do { if (VERBOSE) logerror x; } while (0)
```

Comment:
You've done this in a lot of files, but I have to ask why.  So, why do you make your macros require the double parentheses?  Why not do something like:
```
#define LOGPRINT(...) do { if (VERBOSE) logerror(__VA_ARGS__); } while (0)
```

This would allow it to be invoked with more natural syntax:
```
LOGPRINT("format", arg1, arg2);
```
```

The double parentheses in the macro invocations look very strange and make the code harder to read.  It also allows weird injection of additional things into the macro, e.g. what would happen if you did something like this with the current code:
```
LOGPRINT(("boo!"), some_call());
```